packages/core/src/utils/transformers.ts

```diff
@@ -39,6 +41,29 @@ export function RecordTransformer<T>(Class: { new (...args: any[]): T }) {
     }
   })
 }
+
+/*
+ * Decorator that transforms to and from a metadata instance.
+ *
+ */
+export function MetadataTransformer() {
+  return Transform(({ value, type }) => {
+    switch (type) {
+      case TransformationType.CLASS_TO_PLAIN:
+        return { ...value.data }
+
+      case TransformationType.PLAIN_TO_CLASS:
+        // TODO: Remove conversion in 0.1.0 via a migration script
+        JsonTransformer.renameKey('requestMetadata', 'indyRequestMetadata', value)
+        JsonTransformer.renameKey('credentialMetadata', 'indyCredentialMetadata', value)
+
+        return new Metadata(value)
```

Comment:
The previous structure of the metadata object in `CredentialRecord` was as follows:

```typescript
{
    requestMetadata?: Record<string, unknown>
    credentialDefinitionId?: string
    schemaId?: string
}
```

While it now should look something like the following:

```typescript
{
   indyRequestMetadata?: Record<string, unknown>
   indyCredentialMetadata?: {
      schemaId?: string
      credentialDefinitionId?: string
  }
]
```

It is fine to rename `requestMetadata` to `indyRequestMetadata`, but renaming `credentialMetadata` to `indyCredentialMetadata` is not going to do the trick.

I suggest doing something like the following:

```suggestion
      case TransformationType.PLAIN_TO_CLASS:
      
	  if (value.schemaId || value.credentialDefinitionId || value.requestMetadata) {
		  const metadata = new Metadata({})
		  
		  if (value.schemaId) metadata.add('indyCredentialMetadata', { schemaId: value.schemaId })
		  if (value.credentialDefinitionId) metadata.add('indyCredentialMetadata', { schemaId: value.credentialDefinitionId })
		  if (value.requestMetadata) metadata.set('indyRequestMetadata', value.requestMetadata)
	  
		  return metadata
	  }
	  
	return new Metadata(value)
```

Reply:
Also would be good to add a test that checks if the transformation was successful